[DOCS-3404] Move custom metrics docs into new custom metrics folder

maycmlee opened the pull request:
### What does this PR do?

- Moves 4 custom metrics docs into the custom folder
- Adds alias to front matter for each doc
- Updates navbar URL for each doc
- Updates references to the old links with the new links
- Makes same structural change for all the images
- Updates img src path
- Removes an image that's not being used 

### Motivation

DOCS-3404

### Preview

These links should work, and there should be no broken images in the `type_modifiers` and `dogstatsd_metrics_submission` pages (there are no images on the other two pages):
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/custom_metrics/type_modifiers
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/custom_metrics/agent_metrics_submission
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/custom_metrics/dogstatsd_metrics_submission
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/custom_metrics/powershell_metrics_submission

These links should redirect correctly:
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/type_modifiers
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/agent_metrics_submission
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/dogstatsd_metrics_submission
https://docs-staging.datadoghq.com/may/custom_metrics_folder/metrics/powershell_metrics_submission

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

jtappa reviewed (changes requested):
We should also make the same structural change for all of the images on these pages in the static/images directory & their img src paths

maycmlee commented:
Thanks @jtappa for catching that! I made the structural changes to the images folder and updated the img src paths. I also deleted an image that wasn't being used in the agent_metrics_submission page. Let me know if I should create an editorial card for this because the changes have added up 😅
